### DISCUSSION TITLE
Renovate fails to update dockerfile dependency `FROM docker.io/renovate/renovate:37.244.0-slim` when used with `config:recommended` preset

### DISCUSSION THREAD

**Discussion by scot-t** [renovatebot/renovate#27980] [Request Help] [answered]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitea / Renovate 37.246.1

### Please tell us more about your question or problem

MRE: <https://github.com/scot-t/renovate-mre-slim-issue>
Expected behaviour: tag should be updated to latest version (currently 37.246.1)
Actual behaviour: no version upgrade found

Observations:
* The behaviour seems to be caused by rules defined in `replacements:containerbase`, introduced in #26128. I'm unclear whether the renovate is intending to remove the '-slim' suffix altogether, or simply ignore it when checking versions.
* This was quite difficult to diagnose - with debug logging enabled, it is not very clear how the index lookup / version selection occurs. Perhaps this could be made more verbose in future? I am fairly new to renovate, could just be unfamiliarity.
* Tested with and without pinDigest (https://github.com/scot-t/renovate-mre-slim-issue/pull/1), no version update is found either way.
* Identified with self-hosted, mend renovate app shows same behaviour in mre
* This looks like the same issue reported in https://github.com/renovatebot/renovate/discussions/26659

Thanks for this awesome tool!

**Comment by rarkins** [comment]
@viceice here is the applicable rule: https://github.com/renovatebot/renovate/blob/737375c26077308d505903be96daec15360884fe/lib/config/presets/internal/replacements.ts#L167-L174

Is it intentionally missing a replacementVersion field?

Truly minimal reproduction is here: https://github.com/renovate-reproductions/27980

**Reply by viceice** [reply]
🤔 i thought this rule simply drops the slim suffix. Will do some debugging now

**Reply by rarkins** [reply]
Let's drop this preset temporarily if there's no quick fix, seems it might block people upgrading

**Reply by viceice** [reply]
no, found it https://github.com/renovate-reproductions/27980/commit/7111443be47076677df971b146ff1ce2d85dc183

- https://github.com/renovate-reproductions/27980/pull/1

sending a pr now.

**Reply by viceice** [reply, marked as answer]
- https://github.com/renovatebot/renovate/pull/27991
